src/routes/safe/components/Apps/components/SafeAppCard/SafeAppCard.tsx

```diff
@@ -0,0 +1,238 @@
+import { SyntheticEvent } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { Link } from 'react-router-dom'
+import styled from 'styled-components'
+import { motion } from 'framer-motion'
+import IconButton from '@material-ui/core/IconButton'
+import { alpha } from '@material-ui/core/styles/colorManipulator'
+import { Card, Title, Text, Icon } from '@gnosis.pm/safe-react-components'
+
+import { extractSafeAddress, generateSafeRoute, getShareSafeAppUrl, SAFE_ROUTES } from 'src/routes/routes'
+import { SafeApp } from 'src/routes/safe/components/Apps/types'
+import fallbackSafeAppLogoSvg from 'src/assets/icons/apps.svg'
+import { currentChainId } from 'src/logic/config/store/selectors'
+import enqueueSnackbar from 'src/logic/notifications/store/actions/enqueueSnackbar'
+import { enhanceSnackbarForAction, NOTIFICATIONS } from 'src/logic/notifications'
+import { FETCH_STATUS } from 'src/utils/requests'
+import { copyToClipboard } from 'src/utils/clipboard'
+import { getShortName } from 'src/config'
+import { SafeAppDescriptionSK, SafeAppLogoSK, SafeAppTitleSK } from './SafeAppSkeleton'
+
+type SafeAppCardSize = 'md' | 'lg'
+
+type SafeAppCardProps = {
+  safeApp: SafeApp
+  size: SafeAppCardSize
+  togglePin: (app: SafeApp) => void
+  isPinned?: boolean
+  isCustomSafeApp?: boolean
+  onRemove?: (app: SafeApp) => void
+}
+
+const SafeAppCard = ({
+  safeApp,
+  size,
+  togglePin,
+  isPinned,
+  isCustomSafeApp,
+  onRemove,
+}: SafeAppCardProps): React.ReactElement => {
+  const chainId = useSelector(currentChainId)
+  const dispatch = useDispatch()
+
+  const safeAddress = extractSafeAddress()
+  const appsPath = generateSafeRoute(SAFE_ROUTES.APPS, {
+    shortName: getShortName(),
+    safeAddress,
+  })
```

Comment:
Before merging, let's use this hook (https://github.com/safe-global/safe-react/pull/3826) or I'll merge main into that PR and edit it there (depending on which one makes it through QA first).

Please leave this comment unresolved.